rust/agents/relayer/src/prover.rs

```diff
@@ -77,19 +85,29 @@ impl Prover {
     /// Create a proof of a leaf in this tree.
     ///
     /// Note, if the tree ingests more leaves, the root will need to be recalculated.
-    pub fn prove(&self, index: usize) -> Result<Proof, ProverError> {
+    pub fn prove_against_latest(&self, index: usize) -> Result<Proof, ProverError> {
         if index > u32::MAX as usize {
             return Err(ProverError::IndexTooHigh(index));
         }
         let count = self.count();
         if index >= count {
             return Err(ProverError::ZeroProof { index, count });
         }
+        Ok(get_proof(&self.tree, index))
+    }
 
-        let (leaf, hashes) = self.tree.generate_proof(index, TREE_DEPTH);
-        let mut path = [H256::zero(); 32];
-        path.copy_from_slice(&hashes[..32]);
-        Ok(Proof { leaf, index, path })
+    #[instrument(err, skip(self), fields(prover_msg_count=self.count()))]
+    pub fn prove_against_historic(
```

Comment:
Wondering what you think of putting this logic into the MerkleTree impl https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1484#discussion_r1059390358